src/test/java/org/apache/maven/shared/utils/cli/CommandLineUtilsTest.java

```diff
@@ -195,6 +190,18 @@ public void environmentVariableFromSystemIsCopied() {
         assertThat(environmentVariables, hasItemInArray( "TEST_SHARED_ENV=TestValue" ) );
     }
 
+    @Test
+    public void environmentVariableFromSystemIsNotCopiedIfInheritedIsFalse() {
+
+        Commandline commandline = new Commandline();
+        commandline.setShellEnvironmentInherited( false );
+
+        String[] environmentVariables = commandline.getEnvironmentVariables();
+
+        assertNotNull(environmentVariables);
+        assertThat(environmentVariables, emptyArray() );
```

Comment:
assertEquals(0, environmentVariables.length)

would be much easier to read here.